sympy/core/numbers.py

```diff
@@ -211,6 +211,7 @@ def igcd(*args):
 except ImportError:
     def igcd2(a, b):
         """Compute gcd of two Python integers a and b."""
+        a, b = as_int(a), as_int(b)
```

Comment:
`igcd2` is not called in many places. I think the obvious suspects are the calls after `if isinstance(other, integer_types)` in `pythonrational`.